lib/src/ast_extensions.dart

```diff
@@ -126,17 +126,61 @@ extension ExpressionExtensions on Expression {
   ///   element,
   /// ];
   /// ```
-  bool get isDelimited => switch (this) {
-        FunctionExpression() => true,
-        InstanceCreationExpression() => true,
-        ListLiteral() => true,
-        MethodInvocation() => true,
-        ParenthesizedExpression(:var expression) => expression.isDelimited,
-        RecordLiteral() => true,
-        SetOrMapLiteral() => true,
-        SwitchExpression() => true,
-        _ => false,
-      };
+  ///
+  /// Likewise, in an argument list, delimited expressions can be given
+  /// block-like formatting:
+  ///
+  /// ```
+  /// function([
+  ///   element,
+  /// ]);
+  /// ```
+  bool get isDelimited {
+    // Unwrap named expressions to get the real expression inside.
+    var expression = this;
+    if (expression is NamedExpression) {
+      expression = expression.expression;
+    }
+
+    // TODO(tall): We should also allow multi-line strings to be formatted
+    // like block arguments, at least in some cases like:
+    //
+    // ```
+    // function('''
+    //   Lots of
+    //   text
+    // ''');
+    // ```
+
+    // TODO(tall): Consider whether immediately-invoked function expressions
+    // should be block argument candidates, like:
+    //
+    // ```
+    // function(() {
+    //   body;
+    // }());
+    // ```
+    return switch (expression) {
+      // A function expression can use either a non-empty parameter list or a
+      // non-empty block body for block formatting.
+      FunctionExpression(:var parameters!, :var body) =>
```

Comment:
Is the comment intended to explain the `!`? I don't think I understand it.

Reply:
I changed it to `?`. In practice it doesn't matter. The `parameters` field in `FunctionExpression` is nullable because `FunctionExpression` is used for both anonymous functions and as an inner AST node for function declarations. In the latter case, you might be defining a getter where there is no parameter list, so `parameters` is nullable.

Dart doesn't have anonymous getters, so any actual function expression will have a `parameters`. And we only ever call `isDelimited` on expressions, not declarations. So in practice, `parameters` will always be present.

But... it's not hard to define `isDelimited` to correctly handle an absent parameter list. In that case, it can't split at the parameter list. So I just did that.